sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/PredicateSuite.scala

```diff
@@ -126,6 +126,19 @@ class PredicateSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkEvaluation(IsNaN(Literal(5.5f)), false)
   }
 
+  test("NaN equality and comparison") {
+    def testNaN(nan: Expression): Unit = {
+      checkEvaluation(nan === nan, true)
+      checkEvaluation(nan <=> nan, true)
+//      checkEvaluation(nan <= nan, true)
```

Comment:
Ah, I also see now that I should remove this test and add `NaN` literals to the `equalValues` list above.